joblib/test/test_parallel.py

```diff
@@ -2122,3 +2116,58 @@ def parallel_call(n_jobs):
         parallel_call(n_jobs)
         executor = get_reusable_executor(reuse=True)
         assert executor == first_executor
+
+
+def _set_initialized(initialized):
+    import os
+
+    initialized[os.getpid()] = True
+
+
+def _check_initialized(initialized):
+    time.sleep(0.01)
+    return initialized.get(os.getpid(), False)
+
+
+@with_multiprocessing
+@parametrize("n_jobs", [2, 4, -1])
+@parametrize("backend", PROCESS_BACKENDS)
+@parametrize("context", [parallel_config, parallel_backend])
+def test_initializer(n_jobs, backend, context):
+    n_jobs = effective_n_jobs(n_jobs)
+    manager = mp.Manager()
+    initialized = manager.dict()
+
+    with context(
+        backend=backend,
+        n_jobs=n_jobs,
+        initializer=_set_initialized,
+        initargs=(initialized,),
+    ):
+        with Parallel() as parallel:
+            result = parallel(
+                delayed(_check_initialized)(initialized) for i in range(n_jobs)
+            )
+
+        assert all(result)
+
+
+@with_multiprocessing
+@pytest.mark.parametrize("n_jobs", [2, 4, -1])
+def test_initializer_reuse(n_jobs):
+    # test that the initializer is called only once
+    # when the executor is reused
```

Comment:
@tomMoral this comment seems off.

Maybe the following better reflects what's happening.

```suggestion
    # Check that it is possible to pass initializer config via the `Parallel`
    # call directly. Each parallel call set its own initializer args,
    # independently of the previous calls, hence the loky workers are not
    # reused.
```

But we should also add a case where they are reused: the `initilized` dict should be defined outside the for loop.

We should further inspect the pids in the `initialized` dict from the main process to check that this is actually the case (the number of distinct pids matches the value of `n_jobs`) probably in a retry-sleep-timeout-loop.